src/dune_engine/sandbox.ml

```diff
@@ -116,6 +135,22 @@ let create ~mode ~rule_loc ~deps ~rule_dir ~rule_digest ~expand_aliases =
   (* CR-someday amokhov: Note that this doesn't link dynamic dependencies, so
      targets produced dynamically will be unavailable. *)
   link_deps t ~mode ~deps;
+  let () =
+    match (common_metrics, dune_stats) with
+    | Some _, None | None, Some _ -> assert false
+    | None, None -> ()
+    | Some common, Some dune_stats ->
+      let now = Unix.gettimeofday () in
```

Comment:
All this messing with times looks horrible. Can't we move it out to some helper module, not to pollute the sandboxing logic?

Reply:
(I'd expect the diff in this module to be 2-3 lines, not 37!)

Reply:
Couldn't reduce to 2-3, but hopefully this is nimble enough.

Reply:
Thanks, that's much better!